primitives/account/src/staking_contract/staker.rs

```diff
@@ -35,7 +36,7 @@ impl StakingContract {
         staker_address: &Address,
         value: Coin,
         delegation: Option<Address>,
-    ) -> Result<(), AccountError> {
+    ) -> Result<Vec<Log>, AccountError> {
```

Comment:
I think this could also return 
```suggestion
    ) -> Result<AccountInfo, AccountError> {
```
Same reasoning as for `stake`.